internal/pkg/agent/application/upgrade/cleanup.go

```diff
@@ -13,11 +13,15 @@ import (
 	"github.com/hashicorp/go-multierror"
 
 	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
 )
 
-// preUpgradeCleanup will remove files that do not have the passed version number from the downloads directory.
-func preUpgradeCleanup(version string) error {
-	files, err := os.ReadDir(paths.Downloads())
+// cleanNonMatchingVersionsFromDownloads will remove files that do not have the passed version number from the downloads directory.
+func cleanNonMatchingVersionsFromDownloads(log *logger.Logger, version string) error {
+	downloadsPath := paths.Downloads()
+	log.Infow("Cleaning up non-matching downloaded versions", "version", version, "downloads.path", downloadsPath)
```

Comment:
we were trying to reduce amount of info level logs, consider debug

Reply:
I don't think we should do this around upgrades. Upgrade failures are some of the worst errors we can experience, and for failures that are hard to reproduce or leave the agent in a poor state you sometimes do not get the chance to go back and enable debug logs.